### PR TITLE
Issue 745 - Rename async modules

### LINKED ISSUE #745: Rename async modules
My company uses a centralized access management system to validate a user is authorized to view websites. The system rejects any requests going to URL's containing the tilde character. This prevents all dependencies with names like async~xxx from being downloaded when the browser requests them.

Is it possible to replace the tilde with a different character such as underscore ( _ )?

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
Closes #745

### CODE REVIEW

File: dash_core_components_base/__init__.py [hunk 1]
```diff
@@ -116,19 +116,19 @@
         'async': 'eager'
     },
     {
-        'relative_package_path': 'async~plotlyjs.js',
+        'relative_package_path': 'async-plotlyjs.js',
         'external_url': (
             'https://unpkg.com/dash-core-components@{}'
-            '/dash_core_components/async~graph~plotlyjs.js'
+            '/dash_core_components/async-plotlyjs.js'
```
Comment (Marc-Andre-Rivet) [inline dash_core_components_base/__init__.py]: Pretty bad error here and below with `serve_locally=False`

Comment (alexcjohnson) [reply dash_core_components_base/__init__.py]: Yikes, nice find! Is there anything we can do to test this? I'm thinking like a standard test we can put in `dash.testing` and then run in all our repos with appropriate parameters, looping over `_js_dist` and ensuring `external_url` matches a certain pattern vs `relative_package_path`?

Comment (alexcjohnson) [reply dash_core_components_base/__init__.py]: a followup issue would be fine for now.

Comment (Marc-Andre-Rivet) [reply dash_core_components_base/__init__.py]: Follow up in https://github.com/plotly/dash-core/issues/171 - providing at least two approaches: pattern matching as suggested and something more robust / matching e2e usage but requiring more effort.

**Comment by ManojSN** [comment]
Team, Thank you for changing the ~ to a -. The use @ sign in the URLS is may also not be considered safe and be blocked. 

Would it be possible to alter that too?

 'external_url': (
            'https://unpkg.com/dash-core-components@{}'
            '/dash_core_components/dash_core_components.min.js'
        ).format(__version__)

**Comment by alexcjohnson** [comment]
> The use @ sign in the URLS is may also not be considered safe and be blocked.

That's just the way https://unpkg.com/ works, we have no control over that. But your app will only use access those URLs if you explicitly set `serve_locally=False`

**Pull request merged**